### DISCUSSION TITLE
No IntelliSense in VSCode for Prisma Client

### DISCUSSION THREAD

**Discussion by ggukggukee** [prisma/prisma#21888] [Q&A]
### Bug description

I have Next.js fullstack application and there is my prisma client's code

```js
import { PrismaClient } from '@prisma/client';

let prisma;

if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}

export default prisma;
```

if i use prisma client like that i don't have query autocompletion. IntelliSense is not working in VSCode.
When i change it to:
```js
const prisma = new PrismaClient();

export default prisma;
```
it's working as intended.

But my application is running on serverless environment therefore i have to use code above. Therefore I don't change my prisma client code every time when I code and change it before pushing to production.

I'm using JavaScript

### How to reproduce

1. Add prisma client code
2. No Prisma's IntelliSense is shown

### Prisma information

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextSearch"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

### Environment & setup

- OS:  MacOS 14.1
- Database: PostgreSQL 15.1
- Node.js version: 18.16.1

### Prisma Version

```
prisma                  : 4.16.2
@prisma/client          : 4.16.2
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 4bc8b6e1b66cb932731fb1bdbbc550d1e010de81 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 4bc8b6e1b66cb932731fb1bdbbc550d1e010de81 (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.16.1-1.4bc8b6e1b66cb932731fb1bdbbc550d1e010de81
Default Engines Hash    : 4bc8b6e1b66cb932731fb1bdbbc550d1e010de81
Studio                  : 0.484.0
Preview Features        : fullTextSearch
```

**Comment by ludralph** [comment]
Hi @ggukggukee 👋 

Thank you for raising this question.

Did you try restarting the Typescript server in VSCode? To restart the server, you can use the commands below
```
Mac: Cmd + Shift + P
Win: Ctrl + Shift + P
Then type restart and the option should be available.
```
Do let me know if this helps to resolve the issue.

**Reply by ggukggukee** [reply]
I don't use typescript, just javascript. if I use prisma like this in development:
```js
const prisma = new PrismaClient();

export default prisma;
```
I have IntelliSense but I have warning that multiple prisma instances are created.
When I use Prisma Client with `global.prisma` I have no IntelliSense

**Reply by NisuSan** [reply]
@ludralph, I aslo faced with problem, when IntelliSense not working in **takeover** mode in Nuxt3 project.
My code is:
```javascript 
import { PrismaClient } from '@prisma/client'

const globalForPrisma = global as unknown as {
  prisma: PrismaClient | undefined
}

export const prisma = globalForPrisma.prisma ?? new PrismaClient()
if (process.env.NODE_ENV !== 'production') globalForPrisma.prisma = prisma
``` 

Also I noteced, that type of `PrismaClient` is **any**
![Screenshot 2023-11-21 164158](https://github.com/prisma/prisma/assets/14041642/5ea9dde5-4b27-4293-9999-3d7dfeadd0d7)

Versions of the packages:
```
"@prisma/client": "^5.6.0",
"prisma": "^5.6.0",
```

**Reply by Satindar31** [reply]
@ludralph Facing the same issue with NextJS 14. On version 5.12.1

**Reply by akowalska622** [reply]
Worked for me though facing the same issue, but using TypeScript

**Reply by devesh-kmishra** [reply]
@ggukggukee Did you find any solution for this problem?
Edit: If anyone faces this problem in the future, make sure to `import { PrismaClient } from "@/generated/prisma";
` or wherever the prisma client has been generated in your application, and not from `@prisma/client`.

**Reply by mosfetnik** [reply]
Thanks , mine will work now

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
